sydent/http/servlets/__init__.py

```diff
@@ -39,25 +39,32 @@ def get_args(request, required_args):
     """
     Helper function to get arguments for an HTTP request.
     Currently takes args from the top level keys of a json object or
-    www-form-urlencoded for backwards compatability.
+    www-form-urlencoded for backwards compatability on v1 endpoints only.
     Returns a tuple (error, args) where if error is non-null,
     the request is malformed. Otherwise, args contains the
     parameters passed.
     """
+    v1_path = request.path.startswith('/_matrix/identity/v1')
+
     args = None
+    # for v1 paths, only look for json args if content type is json
     if (
-        request.requestHeaders.hasHeader('Content-Type') and
-        request.requestHeaders.getRawHeaders('Content-Type')[0].startswith('application/json')
+        request.method == 'POST' and (
```

Comment:
Why are we checking for post?

Reply:
'cos this is determining if we're getting args from json, which we can only do for POST requests (as opposed to GET - I don't think sydent has any PUTs)